src/zope/app/wsgi/tests.py

```diff
@@ -12,16 +12,22 @@
 #
 ##############################################################################
 """WSGI tests"""
+from zope.app.wsgi.testing import SillyMiddleWare
+from zope.app.wsgi.testlayer import BrowserLayer
+from zope.authentication.interfaces import IUnauthenticatedPrincipal
+from zope.component.testlayer import ZCMLFileLayer
+from zope.publisher.interfaces.logginginfo import ILoggingInfo
+from zope.testing import renormalizing
 import doctest
+import io
 import re
 import unittest
 import zope.app.wsgi
-import zope.event
+import zope.component
 import zope.component.testing
-from zope.app.wsgi.testing import SillyMiddleWare
-from zope.app.wsgi.testlayer import BrowserLayer
-from zope.component.testlayer import ZCMLFileLayer
-from zope.testing import renormalizing
+import zope.event
+import zope.interface
+
```

Comment:
Please don't reorder 3rd-party imports above stdlib imports.

(A blank line between the two import sections would not be amiss.)